src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -185,6 +189,40 @@ private function extractFromAccessor(string $class, string $property): ?array
         return null;
     }
 
+    /**
+     * Tries to extract type information from constructor.
+     *
+     * @return Type[]|null
+     */
+    private function extractFromConstructor(string $class, string $property): ?array
```

Comment:
why `?array`, and not `?Type` ?

Reply:
Because it returns an array a not a Type